### PR TITLE
New module for associating files with phrases - assocfile

### PR DESCRIPTION
**Pull request by p0358**
Based on "associate" module, lets associate files like images or videos
to be sent into a chat with phrases. It's looking for files inside
"files" folder in Kassy's main directory.

### CODE REVIEW

**Comment by mrkno** [comment]
@p0358 our long term plan is to move all the modules out into their own git repositories so they can be `kpm install`ed. There are a few options here: we can merge this in to the project and move it out later with the rest of the ones that are still in there ([there are a number which arn't](https://github.com/mrkno/Kassy/wiki/KPM-Table)), create a new repository under the https://github.com/concierge organisation for it (where we intend to move the modules to) or you can maintain a personal repository for it.

Which would you prefer?

File: modules/assocfile/assocfile.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			if (api) {
+				api.sendFile("file", "files\\"+this.config[thread][assoc], "", thread);
+			}
+			return true;
+		}
+	}
+	return false;
+};
+
+exports.toggleAssociation = function(thread, hook, text) {
```
Comment (mrkno) [inline modules/assocfile/assocfile.js]: Nothing other than `help`, `run` and `match` should be exported. In a soon to be merged branch, the `associate

Comment (mrkno) [inline modules/assocfile/assocfile.js]: Other than this, the code you have looks good.

**Comment by p0358** [comment]
Yes, I was testing it even additionally now and...
It didn't want to send files with sendFile to Facebook, sendImage works for video and audio files as well, but it didn't work for text file I tried to send for some reason... Additionally corrected some messages output.
I will update exports after I see how it was done in the "associate", because something didn't want to work after I tried, so I gave up. 
We can can merge this in to the project for now and move it out later with the rest of the ones that are still in there.
I think KPM modules list should be linked somewhere in main readme.

**Comment by mrkno** [comment]
Approved. Will require changes for Hubot in future anyway so it is easier to do that in the hubot branch at the same time as everything else.

**Pull request merged**